File: backend/app/routers/tasks.py
"""任务中心路由：任务列表 / 详情（进度轮询）/ 重试。"""
from __future__ import annotations

from fastapi import APIRouter, Depends
from sqlalchemy.orm import Session

from app.common.response import ApiError, ok
from app.database import get_db
from app.deps import get_current_user, require_permission
from app.models import Task, User
from app.serializers import task_to_dict
from app.services import indexing

router = APIRouter(prefix="/tasks", tags=["tasks"])


@router.get("")
def list_tasks(
    db: Session = Depends(get_db), user: User = Depends(require_permission("admin"))
):
    tasks = db.query(Task).order_by(Task.created_at.desc()).all()
    return ok([task_to_dict(t) for t in tasks])


@router.get("/{task_id}")
def get_task(
    task_id: str, db: Session = Depends(get_db), user: User = Depends(get_current_user)
):
    """任务进度查询：供上传/重建后前端轮询，登录用户均可查询。"""
    task = db.get(Task, task_id)
    if not task:
        raise ApiError("任务不存在", code=1, status_code=404)
    return ok(task_to_dict(task))


@router.post("/{task_id}/retry")
def retry_task(
    task_id: str,
    db: Session = Depends(get_db),
    user: User = Depends(require_permission("admin")),
):
    task = db.get(Task, task_id)
    if not task:
        raise ApiError("任务不存在", code=1, status_code=404)
    if not task.doc_id:
        raise ApiError("该任务不支持重试", code=1, status_code=400)
    task.retry_count += 1
    task.state = "PENDING"
    task.progress = 0
    task.message = ""
    db.commit()
    new_task_id = indexing.create_index_task(
        task.doc_id, task_type=task.type, target=task.target
    )
    return ok({"taskId": new_task_id}, message="已重新提交任务")

File: frontend/src/mock/tasks.ts
import type { TaskItem } from '../types'

export const tasks: TaskItem[] = [
  { taskId: 'celery-001', type: '文档索引', target: '新版假期政策（草案）.docx', state: 'PROGRESS', progress: 65, retryCount: 0, message: '正在向量化中...', createdAt: '2026-07-24 09:00:00' },
  { taskId: 'celery-002', type: '文档索引', target: '报销单模板说明.html', state: 'PROGRESS', progress: 82, retryCount: 0, message: '正在向量化中...', createdAt: '2026-07-24 10:30:00' },
  { taskId: 'celery-003', type: '文档索引', target: '考勤管理办法 v3.2.pdf', state: 'SUCCESS', progress: 100, retryCount: 0, message: '索引完成', createdAt: '2026-07-22 14:31:00' },
  { taskId: 'celery-004', type: '重建索引', target: '产品 A 用户手册 v4.0.pdf', state: 'SUCCESS', progress: 100, retryCount: 0, message: '重建索引完成', createdAt: '2026-07-23 10:00:00' },
  { taskId: 'celery-005', type: '文档索引', target: '外派人员考勤.pdf', state: 'FAILURE', progress: 42, retryCount: 3, message: 'PDF 解析失败：格式损坏', createdAt: '2026-07-24 08:05:00' },
  { taskId: 'celery-006', type: '文档索引', target: '数据库设计规范.pdf', state: 'SUCCESS', progress: 100, retryCount: 0, message: '索引完成', createdAt: '2026-07-18 15:00:00' },
  { taskId: 'celery-007', type: '批量评测', target: '技术规范数据集', state: 'SUCCESS', progress: 100, retryCount: 0, message: '评测完成', createdAt: '2026-07-21 17:00:00' },
  { taskId: 'celery-008', type: '批量评测', target: '产品手册质量验证', state: 'PROGRESS', progress: 45, retryCount: 0, message: '第 3/8 批次执行中...', createdAt: '2026-07-24 11:00:00' },
  { taskId: 'celery-009', type: '文档索引', target: '前端 React 编码规范.md', state: 'SUCCESS', progress: 100, retryCount: 0, message: '索引完成', createdAt: '2026-07-20 13:00:00' },
  { taskId: 'celery-010', type: '文档索引', target: '微服务架构评审模板.docx', state: 'PENDING', progress: 0, retryCount: 0, message: '等待执行', createdAt: '2026-07-24 11:15:00' },
]
